src/cargo/util/toml.rs

```diff
@@ -564,15 +597,71 @@ fn inferred_bench_targets(layout: &Layout) -> Vec<TomlTarget> {
 }
 
 impl TomlManifest {
-    fn to_real_manifest(&self,
+    pub fn prepare_for_publish(&self) -> TomlManifest {
+        return TomlManifest {
+            package: self.package.clone(),
+            project: self.project.clone(),
+            profile: self.profile.clone(),
+            lib: self.lib.clone(),
+            bin: self.bin.clone(),
+            example: self.example.clone(),
+            test: self.test.clone(),
+            bench: self.bench.clone(),
+            dependencies: map_deps(&self.dependencies),
+            dev_dependencies: map_deps(&self.dev_dependencies),
+            dev_dependencies2: map_deps(&self.dev_dependencies2),
```

Comment:
Perhaps we may normalize `dev_dependencies2` into `dev_dependencies` at this step?

Even if we don't plan to drop support for the second variant, having only one form may be useful if, for example, someone greps `~/.cargo` or otherwise interacts with crates without Cargo itself.